File: FS-Intro/src/StartingWithFSharp/FPDemoCS/Dollars.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace FPDemoCS
{
    public class Dollars
    {
        private readonly decimal _amount;
        public Dollars(decimal value)
        {
            _amount = value;
        }
        public decimal Amount
        {
            get { return _amount; }
        }
        public Dollars Times(decimal multiplier)
        {
            return new Dollars(this._amount * multiplier);
        }
    }

}

File: FS-Intro/src/StartingWithFSharp/InteropLib/Dollars.fs
namespace InteropLib

[<CLIMutableAttribute>]
type DollarsFS = 
    { Amount : decimal }
    member x.Times(multiplier) = { Amount = x.Amount * multiplier }

//let d1 = { Amount = 10M }
//let d2 = { Amount = 5M }
//
//d1 = d2.Times(2M)
